### ISSUE TITLE
GUI is almost unreadable when using dark Qt themes

### ISSUE THREAD

**Issue by sq5gvm** [fwbuilder/fwbuilder#28]
When using dark theme (ie KDE Breeze Dark) parts of the GUI are unreadable (especially object tree but also text in edit fields). See screenshot for details.
![screenshot_20170407_232416](https://cloud.githubusercontent.com/assets/2500520/24820481/837a5f1e-1be9-11e7-8115-4e2e597069d4.png)

**Comment by wolfpackmars2** [comment]
+1 for this. Linux Mint has moved to a dark theme resulting in the same problem

**Comment by karora** [comment]
As I get older and my eyesight gets worse I have found it worthwhile to switch to a dark theme for everything, and support for these themes has improved leaps and bounds, but fwbuilder continues to be extremely difficult to use with it's "white on light-green" and "black on dark-grey" areas.

If someone could point in approximately the right place I'd be happy to work up a patch for this.

**Comment by karora** [comment]
I've made a small change to the src/libgui/utils.cpp to switch to using white text when the background is dark, and added a pull request for that now.

This doesn't change the colours in the Standard or User tree, although that can be changed by manually editing the saved `.fwb` file and changing `#d2ffd0` to something like `#123f10`.  Changing the blue background for the Standard tree can be accomplished by editing the resources `src/res/objects_init.xml.in` (look for `#d4f8ff` and change it to something like `#14383f`) and rebuilding, or perhaps similarly changing `/usr/share/fwbuilder/objects_init.xml` on an installed system.

I'll see if I can provide another patch to allow those values to be set in the program, rather than in the `.fwb` file and/or the `objects_init.xml` file.  Especially the `.fwb` file seems a very odd place to be storing the color!

**Comment by karora** [comment]
Actually as I play with this now I see that the colour is stored in the `.fwb` file I notice that I can right-click on the `User` tree, choose `Edit` from the context menu and change the colour directly in the object edit dialog...

**Comment by gibsonlp** [comment]
I have the same issue on OSX. I tried changing the font colors in the preferences to no avail, is there a quick solution for this? very annoying :/

**Comment by karora** [comment]
@gibsonlp you could try rebuild with my patch applied - see pull request #79 

You can also edit some of the colours directly into the XML of the *.fwb, which is hackish, but relatively quick.

**Comment by gibsonlp** [comment]
@karora 

Thank you. I applied your patch, it fixed some of it but not everything, see attached screenshots:
![Screen Shot 2020-03-18 at 20 40 09](https://user-images.githubusercontent.com/93285/76995870-30cd7b00-6959-11ea-9006-df52a8a9ba49.png)

Any idea what's next? It's crazy that this has been open since 2018 :(
Thanks.

**Comment by sirius** [comment]
Hi, @gibsonlp . I'm sorry for not giving firewall builder more love the last years, but time is a scarce resource. However I've been spending a little more time on firewall builder the last few weeks. My main focus has been to fix the unit tests. I'll push this week. My priority now is to release a 6.0 version, and then try to fix things one by one - if time allows - and push more frequently.

**Comment by karora** [comment]
Yeah.  The crazy thing is that the colours are all written in the file itself, so the required changes seem to be very deep.

**Comment by gibsonlp** [comment]
@sirius 
No need to apologize, thank you for working on this to begin with.

@karora - Legacy stuff, I guess :/

**Comment by karora** [comment]
@sirius I'd be happy to hack on the patch I did previously to make it more generally usable, but I don't really understand the code base so only dug deeply enough to make it work OK for me.  If you want to suggest an approach for a more correct resolution I'd be happy to have a go... :-)

**Comment by cherio** [comment]
Same here. Just started version 5.1 and was immediately taken back by the barely readable dark labels on the dark background. I don't think this version of FWB is usable.

![image](https://user-images.githubusercontent.com/2200569/79586409-0fb69200-809f-11ea-9857-33062b68e32d.png)

**Comment by cherio** [comment]
In my case I fixed the issue. I had previously played with `~/.Xresources` and had some leftovers there. So I simply deleted the file and ran `xrdb -remove`. That made FWB awesome again :)
